### PR TITLE
CM-876 - Pay-in flow [ PayMe ]

### LINKED ISSUE #876: Integrate PayMe - Pay in flow
We need to create endpoints for the frontend to use PayMe as as payments processor.

A separate task when needed will be open to remove Circle

Technical flow can be found here - https://miro.com/app/board/o9J_lgjPSwk=/

### LINKED ISSUE #930: Billing details to backend api
Collect and save billing details from frontend and save in DB

### PR DESCRIPTION
**Pull request by ariel-dao**
Architecture overview:

Router -> Controller -> Service -> [ DB and Providers ]

Router (simple express router, points path to controller)
Controller (in charge of initial request validations and adding anything that services might need)
Service (Actual business logic, uses DB functions and Providers to make things happen)
DB (Raw calls to the database)
Provider (Communicator with the selected payment provider, interactions with the provider's API are defined as functions here)
Business (business logic that is now called directly from a route or controller, but from a service or other)

### CODE REVIEW

File: functions/src/payments/db/payments/getPayments.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import admin from 'firebase-admin';
+
+import { PaymentStatus } from '../../../shared/interfaces/payments/PaymentStatus';
+import { IPayment } from '../../../shared/interfaces/payments/IPayment';
+import { PaymentsCollection } from './index';
+import Query = admin.firestore.Query;
+import { CommonError } from '../../../util/errors';
+
+export interface IGetPaymentsOptions {
```
Comment (SobolevYuri) [inline functions/src/payments/db/payments/getPayments.ts:8]: do we wanna have a definitions of types/interfaces in such a places (with out of importing all of them from the according ./shared)?

File: functions/src/payments/db/payments/updatePayment.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { firestore } from 'firebase-admin';
+import { IPayment } from '../../../shared/interfaces/payments/IPayment';
+import { PaymentsCollection } from './index';
+
+interface IUpdatePaymentOptions {
```
Comment (SobolevYuri) [inline functions/src/payments/db/payments/updatePayment.ts:5]: ...the same, and another few places as well

File: functions/src/util/commonApp.ts [hunk 3]
```diff
@@ -23,11 +23,6 @@ export const commonApp = (router?: express.Router, options?: ICommonAppOptions):
   app.use(sessions);
   app.use(ipMiddleware);
 
-  app.use(bodyParser.json());
```
Comment (SobolevYuri) [inline functions/src/util/commonApp.ts:26]: why did you decide to remove this?

File: functions/src/shared/interfaces/payments/payme/CreateChargePaymentPage.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import * as yup from 'yup';
+import { CURRENCIES } from '../../../enums/CURRENCIES';
+
+export const CreateChargePaymentPageRequestPayloadValidationSchema = yup.object({
```
Comment (SobolevYuri) [inline functions/src/shared/interfaces/payments/payme/CreateChargePaymentPage.ts]: are we going to have such a structure (exporting of schemas, interfaces and types from a single source regarding to the related feature) everywhere in the project, or not?

**Review by SobolevYuri — APPROVED** [review]

**Review by noamso1 — APPROVED** [review]

File: functions/src/payments/db/payments/getPayments.ts (hunk 1 above)
Comment (ariel-dao) [reply functions/src/payments/db/payments/getPayments.ts:8]: we'll use indexes on appropriate folders in the future

File: functions/src/util/commonApp.ts (hunk 3 above)
Comment (ariel-dao) [reply functions/src/util/commonApp.ts:26]: redundant, express gives this out of the box after 4th version~

**Pull request merged**